purchases/src/main/kotlin/com/revenuecat/purchases/utils.kt

```diff
@@ -71,23 +71,21 @@ internal fun JSONObject.parseDates(jsonKey: String): HashMap<String, Date?> {
         val key = it.next()
 
         val expirationObject = getJSONObject(key)
-
-        if (expirationObject.isNull(jsonKey)) {
-            expirationDates[key] = null
-        } else {
-            val dateValue = expirationObject.getString(jsonKey)
-            try {
-                val date = Iso8601Utils.parse(dateValue)
-                expirationDates[key] = date
-            } catch (e: RuntimeException) {
-                throw JSONException(e.message)
-            }
-        }
+        expirationDates[key] = expirationObject.optDate(jsonKey)
     }
 
     return expirationDates
 }
 
+@Throws(JSONException::class) internal fun JSONObject.getDate(jsonKey: String): Date =
+    try {
+        Iso8601Utils.parse(getString(jsonKey))
+    } catch (e: RuntimeException) {
+        throw JSONException(e.message)
+    }
+
+internal fun JSONObject.optDate(jsonKey: String): Date? = takeUnless { this.isNull(jsonKey) }?.getDate(jsonKey)
```

Comment:
The function in `Factories.kt` was:

```
private fun JSONObject.optDate(name: String) =	
    takeUnless { isNull(name) }?.getString(name)?.let {	
        Iso8601Utils.parse(it)	
    }
```

I replaced  `Iso8601Utils.parse(it)` with `getDate(jsonKey)`